osu.Game/Screens/Select/BeatmapInfoWedge.cs

```diff
@@ -108,14 +108,18 @@ void updateState()
                 Depth = Info?.Depth + 1 ?? 0,
             }, newInfo =>
             {
-                updateState();
-                Add(Info = newInfo);
+                if (newInfo.Working == beatmap)
+                {
+                    updateState();
+                    Add(Info = newInfo);
+                }
             });
         }
 
         public class BufferedWedgeInfo : BufferedContainer
         {
             private readonly WorkingBeatmap working;
+            public WorkingBeatmap Working { get { return working; } }
```

Comment:
`public readonly Beatmap { get; private set; }` and remove the private member above.